flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericCatalogTable.java

```diff
@@ -53,13 +53,6 @@ public GenericCatalogTable copy() {
 			getSchema().copy(), new ArrayList<>(getPartitionKeys()), new HashMap<>(getProperties()), getComment());
 	}
 
-	@Override
-	public Map<String, String> toProperties() {
-		// TODO: Filter out ANY properties that are not needed for table discovery.
```

Comment:
Here TODO item says that we need to filter out any properties that are not needed for table factory discovery. One example that I can think of is GenericInMemoryCatalog.FLINK_IS_GENERIC_KEY.

I also remember that we prefixed some properties before storing them in HMS. Should we remove the prefix in toProperties()?

Reply:
the property prefix is taken care of already.

I will remove FLINK_IS_GENERIC_KEY. I don't think we need to come up with a sound list of properties to keep or remove right now, any changes can be made later on demand.

Reply:
btw, I think FLINK_IS_GENERIC_KEY may be gone soon as we are going to use "type" field to define type of the table, like "hive", "csv", "kafka", etc.

Reply:
well, probably not. I don't think there is going to be a generic property called "type"(, which is different from Blink). It's likely we will retain this key even after combining the two.